ngraph/core/src/op/split.cpp

```diff
@@ -36,28 +36,24 @@ bool ngraph::op::v1::Split::visit_attributes(AttributeVisitor& visitor)
 void op::v1::Split::validate_and_infer_types()
 {
     NGRAPH_OP_SCOPE(v1_Split_validate_and_infer_types);
-    const auto data_ps = input_value(0).get_partial_shape();
-    const auto axis_ps = input_value(1).get_partial_shape();
-    const auto axis_et = input_value(1).get_element_type();
-
-    if (axis_ps.rank().is_static())
-    {
-        NODE_VALIDATION_CHECK(this,
-                              axis_ps.rank().get_length() == 0,
-                              "The 'axis' input is expected to be a scalar. Got: ",
-                              axis_ps);
-    }
+    const PartialShape& data_ps = get_input_partial_shape(0);
+    const PartialShape& axis_ps = get_input_partial_shape(1);
+    const element::Type& axis_et = get_input_element_type(1);
 
     NODE_VALIDATION_CHECK(
-        this, axis_et.is_integral(), "The 'axis' input only accepts integral types");
+        this, axis_ps.rank().compatible(0), "'axis' input must be a scalar. Got: ", axis_ps);
+
+    NODE_VALIDATION_CHECK(this,
+                          axis_et.is_integral_number(),
+                          "Element type of 'axis' input must be integer. Got: ",
+                          axis_et);
 
     PartialShape each_output_shape{data_ps};
+    const Rank data_rank = data_ps.rank();
     const auto axis_input = get_constant_from_source(input_value(1));
-    if (axis_input && data_ps.rank().is_static())
+    if (axis_input && data_rank.is_static())
     {
         auto axis = axis_input->cast_vector<int64_t>()[0];
-
-        const auto data_rank = get_input_partial_shape(0).rank();
         axis = ngraph::normalize_axis(this, axis, data_rank);
 
         if (data_ps[axis].is_static())
```

Comment:
As the `num_splits` is an attribute, what do you think about setting at least upper bound for the output dynamic dimension? Should it work with dynamic interval dims?

https://github.com/openvinotoolkit/openvino/blob/358f3bcba418fe7f16d236e0aa7ae4094cc9649c/ngraph/core/src/op/split.cpp#L70-L75

Reply:
May be it make sense to add `operator/` for the `Dimension` class (the interval logic looks appropriate for the division operation if we perform floor rounding). If we add it then we can write `each_output_shape[axis] = dimension_at_axis / m_num_splits;` regardless of the static/dynamic nature of the input.

@jane-intel , what do you think about this?

Reply:
I have added few changes related to @mitruska comment. The behavior now is:

- if the upper bound of the interval is less than `num_splits` throw exception
- if dimension does not have upper bound, only lower bound is divided
- if dimension has lower and upper bounds, both are divided by `num_splits`

Reply:
I would prefer having operator/ defined for `Dimension`. We already have several operations that require it (like Interpolate)